### ISSUE TITLE
Custom Code - priority needed to execute after jQuery

### ISSUE THREAD

**Issue by euene** [elementor/elementor#14266]
## Prerequisites

- [x] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [x] The feature is still missing in the latest stable version of Elementor ( Elementor Pro. )

There is no way to print any code/scripts after the default bundled jQuery core and jQuery UI from WordPress - therefore all Custom Code JavaScript has to be written in plain/vanilla and can't take advantage of any jQuery. 

Custom code written in jQuery will only work for logged-in users since in that scenario jQuery is being enqueued in the `<head>`, but when logged-out - jQuery will be enqueued in the footer and the custom code will always load before jQuery and errors such as `Uncaught ReferenceError: jQuery is not defined at (index):1073` will be generated in Developer Tools console. 

There's no way to add code after any of Elementor's JavaScript files either - as any Custom Code with `"<body> - End"` and a priority of 10 will still be added before Elementor's JavaScript. 

There should be a way to add code after anything else enqueued by WordPress core or Elementor or any other plug-in in order to make use of functions added by these scripts, or to override these scripts.

### How to reproduce

1. Create new Elementor Pro Custom code
- Title: jQuery Test code
- Location: `"<body> - End"`
- Priority 10
- Content: `<script>jQuery('body').addClass('test');</script>`

2. Publish -> Include -> Entire Site - Save & Close

3. Log out from WordPress and go to the homepage

**Result:** `Uncaught ReferenceError: jQuery is not defined at (index):1073`

### Extra information

As an example, [this plug-in](https://wordpress.org/plugins/addfunc-head-footer-code/) has the option to print custom code/scripts at the end of the body, after jQuery and any other script by letting you enter any number as a priority.

To do that - change the "Site-wide Footer Code" in _**/wp-admin/options-general.php?page=aFhfc_options**_ priority to `100` and the same jQuery code would run fine - so it might be useful to look at this plug-in's implementation for ideas.

Maybe the "10" priority in Elementor can be made so that it prints the code after jQuery and any other Elementor script at the end of the body, or higher/lower priority options can be added?

Thanks.

@shilo-ey

**Comment by whitea2** [comment]
I agree this should be fixed. Until then, my workaround is to use window.onload() which ensures jQuery has been loaded:

```
window.onload = function(){
  // jQuery is loaded
  jQuery( document ).ready(function( $ ){
     // document is ready
  });
};
```

**Comment by shay1383** [comment]
> I agree this should be fixed. Until then, my workaround is to use window.onload() which ensures jQuery has been loaded:
> 
> ```
> window.onload = function(){
>   // jQuery is loaded
>   jQuery( document ).ready(function( $ ){
>      // document is ready
>   });
> };
> ```

So this "> window.onload = function(){" is what I should put my JQuery/JS/$  in in order for it to be loaded right in Elementor Custom Code?

**Comment by euene** [comment]
@DorShahar Could this issue be looked into by Elementor's core developers? Thank you.

**Comment by whitea2** [comment]
> > I agree this should be fixed. Until then, my workaround is to use window.onload() which ensures jQuery has been loaded:
> > ```
> > window.onload = function(){
> >   // jQuery is loaded
> >   jQuery( document ).ready(function( $ ){
> >      // document is ready
> >   });
> > };
> > ```
> 
> So this "> window.onload = function(){" is what I should put my JQuery/JS/$ in in order for it to be loaded right in Elementor Custom Code?

Correct

**Comment by shay1383** [comment]
> > > I agree this should be fixed. Until then, my workaround is to use window.onload() which ensures jQuery has been loaded:
> > > ```
> > > window.onload = function(){
> > >   // jQuery is loaded
> > >   jQuery( document ).ready(function( $ ){
> > >      // document is ready
> > >   });
> > > };
> > > ```
> > 
> > 
> > So this "> window.onload = function(){" is what I should put my JQuery/JS/$ in in order for it to be loaded right in Elementor Custom Code?
> 
> Correct

Sorry to bother you again, I tried your code for this snippet and didn't work:

`jQuery(document).ready(function($) {
    $('#fullpage').fullpage({
        //options here
        scrollingSpeed: 1000,
        navigation: true,
        slidesNavigation: true
    });
    //methods
    $.fn.fullpage.setAllowScrolling(true);
});`

How should I add your code to it?

**Comment by whitea2** [comment]
@shay1383, without the line breaks in your snippet, it's hard to tell what is commented and what is not. You should be able to use something like:
```
window.onload = function(){
  jQuery( document ).ready(function( $ ){
    $('#fullpage').fullpage({
      ...
    });
  });
};

**Comment by shay1383** [comment]
Sorry I didn't manage to embed the snippet correctly
I fixed it now, please check

Also, is it required to add these tags in order for the snippet to be read correctly by the Elementor plugin?:

```
<script type="text/javascript"> 
</script>
```

Thank you

**Comment by whitea2** [comment]
Yes, you need the script tags:

```
<script>
window.onload = function(){
  jQuery( document ).ready(function( $ ){
    ...
  });
};
</script>
```

**Comment by orlandoryo** [comment]
So this was reported in **_February 2021_** and no fix yet? I submitted a ticket last week explaining the same issue, and they haven't replied yet. I thought it was a mobile-only problem until I tried incognito mode after reading this... The "Custom Code" feature is really limited in the current state.

**Comment by orlandoryo** [comment]
> I agree this should be fixed. Until then, my workaround is to use window.onload() which ensures jQuery has been loaded:
> 
> ```
> window.onload = function(){
>   // jQuery is loaded
>   jQuery( document ).ready(function( $ ){
>      // document is ready
>   });
> };
> ```

Man, thanks a lot, this workaround is the way to go for now.

**Comment by shilo-ey** [comment]
This issue has been resolved in Elementor Pro v3.5.0

Feel free to update

Thanks!

**Comment by debihazelden** [comment]
I don't think this has been fixed.
